src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewController.swift

```diff
@@ -200,8 +204,51 @@ extension ExposureDetectionViewController {
 			checkButton.isEnabled = true
 		case .manual:
 			footerView.isHidden = false
-			checkButton.setTitle(AppStrings.ExposureDetection.buttonRefresh, for: .normal)
-			checkButton.isEnabled = riskProvider.manualExposureDetectionState == .possible
+
+			let nextRefresh = riskProvider.nextExposureDetectionDate()
+			let now = Date()
+
+			if nextRefresh > now, countdown == nil {
+				scheduleCountdownTimer(to: nextRefresh)
+			} else if let countdown = countdown, countdown.end != nextRefresh {
+				// Make sure to schedule new countdown if the next refresh time has changed.
+				scheduleCountdownTimer(to: nextRefresh)
+			} else if nextRefresh - 1 > now {
+				showCountdownButton(with: time)
+			} else {
+				showActiveRefreshButton()
+			}
 		}
 	}
+
+	private func scheduleCountdownTimer(to end: Date) {
+		countdown = CountdownTimer(countdownTo: end)
+		countdown?.delegate = self
+		countdown?.start()
+	}
+
+	private func showActiveRefreshButton() {
+		self.checkButton.setTitle(AppStrings.ExposureDetection.buttonRefresh, for: .normal)
+		self.checkButton.isEnabled = riskProvider.manualExposureDetectionState == .possible
+	}
+
+	private func showCountdownButton(with time: String? = nil) {
+		guard let time = time else { return }
+		UIView.performWithoutAnimation {
+			self.checkButton.setTitle(String(format: AppStrings.ExposureDetection.refreshIn, time), for: .normal)
+			self.checkButton.isEnabled = riskProvider.manualExposureDetectionState == .possible
```

Comment:
Copy-pasta'ing from above, because this method has the same 'problem'
> It seems odd to me that a method that is called `showCountdownButton()` uses a boolean check to set a button to an enabled or disabled state. Does that make sense to you either?

Reply:
here we can safely disable the button. I adjusted my code accordingly